templates/client-daemonset.yaml

```diff
@@ -267,7 +284,7 @@ spec:
                 - |
                   {{- if .Values.global.tls.enabled }}
                   curl \
-                    --cacert /consul/tls/ca/tls.crt \
+                    -k \
```

Comment:
Note that I've made the readiness check less secure. This was intentional. My thinking is that it's checking on `localhost` within the pod network security boundary, and the traffic within that boundary is trusted, so it's safe to skip validation. If we wanted to provide the CA, it'd significantly complicate client startup because they'd now need to wait for servers to start, which could increase downtime for Connect-enabled apps, for example, during upgrades.